sympy/calculus/util.py

```diff
@@ -344,6 +345,19 @@ def periodicity(f, symbol, check=False):
         except NotImplementedError:
             pass
 
+    if isinstance(f, Abs):
+        arg = f.args[0]
+        new_f = arg
+        if isinstance(arg, sec):
```

Comment:
Could you just recast all but tan and cot to sin?

```
if isinstance(arg, (cos, csc, sec)):
    # for simplicity cast all but tan and cot functions
    # to sin since the periodicity is the same and then
    # we have a single case to check below
    new_f = sin(arg.args[0])
...
    if period is not None and isinstance(new_f, sin):
       # check to see if the half period satisfies the condition
       # that at x->x+pi, new_f is the same or negated
        test = Eq(abs(new_f.subs(symbol, symbol + period/2))), new_f)
        if check and test not in (S.true, S.false):
            raise error
        elif test:
            return period/2
        return period
...
```